src/controller/CHIPDevice.h

```diff
@@ -142,6 +142,9 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
     CHIP_ERROR SendReadAttributeRequest(app::AttributePathParams aPath, Callback::Cancelable * onSuccessCallback,
                                         Callback::Cancelable * onFailureCallback, app::TLVDataFilter aTlvDataFilter);
 
+    CHIP_ERROR SendWriteAttributeRequest(app::WriteClientHandle aHandle, Callback::Cancelable * onSuccessCallback,
+                                         Callback::Cancelable * onFailureCallback);
+
```

Comment:
We need the WriteClient to carry the attribute payload, while read attribute only requires the path.

Reply:
While we also need to reserve enough flexibility to support reading / writing multiple attributes at the same time. The current solution is expose a WriteClientHandle to let the ChipCluster do the works of encoding.

I would say that, the name and field of Write interaction looks more like ReadInteraction, but from the protocol side, and encoding WriteInteraction should be more like InvokeInteraction, since both of them contains path and cluster specific payloads and both of them will have TimedInteraction in the future.

In this case, Write interaction should have similar API as Invoke (command) interaction has.